my_enumerable.rb

```diff
@@ -1,85 +1,156 @@
+      checker = origin.my_select { |x| x.is_a?(type[0]) } unless type[0].instance_of?(Regexp)
+      checker.empty? ? (return false) : (return true)
+    end
+
+    return true unless all?(*type) { |x| !yield(x) }
 
-    !my_all? { |x| !yield(x) }
+    false
   end
 
-  def my_none?
-    return to_enum(_method_) unless block_given?
+  def my_none?(*type)
+    origin = to_a
+    return true if origin == []
 
-    !my_any? { |x| yield(x) }
+    if !block_given? and type.length.zero?
+      return true if origin.include?(nil) || origin.include?(false)
+
+      return false
+    end
+
+    if type.length == 1
+      checker = origin.my_select { |x| x =~ type[0] } if type[0].instance_of?(Regexp)
+      checker = origin.my_select { |x| x.is_a?(type[0]) } unless type[0].instance_of?(Regexp)
+      checker.empty? ? (return true) : (return false)
+    end
+
+    my_any?(*type) { |x| !yield(x) }
   end
 
   def my_count(argument = nil)
+    origin = to_a
     if argument
-      count = my_select { |x| x == argument }
+      count = origin.my_select { |x| x == argument }
       return count.length
 
     elsif block_given?
-      count = my_select { |x| yield(x) }
+      count = origin.my_select { |x| yield(x) }
       return count.length
     end
 
     length
   end
 
-  def my_map(proc = nil)
-    return to_enum(_method_) unless block_given?
+  def my_map(my_proc = nil)
+    return to_enum(__method__) unless block_given?
 
+    origin = to_a
     array = []
-    if proc
-      my_each { |x| array << procedure.call(x) }
+    if my_proc
+      origin.my_each { |x| array << my_proc.call(x) }
+      return array
     else
-      my_each { |x| array << yield(x) }
+      origin.my_each { |x| array << yield(x) }
     end
     array
   end
 
-  def my_inject(number = nil, operator = nil)
-    if !number.nil?
-      operator = number
-      number = nil
-      to_a.my_each { |item| number = number.nil? ? item : number.send(operator, item) }
-    elsif !operator.nil?
-      to_a.my_each { |item| number = number.nil? ? item : number.send(operator, item) }
-    else
-      to_a.my_each { |item| number = number.nil? ? item : yield(number, item) }
+  def my_inject(*arg)
+    origin = to_a
+    if !block_given? && arg.length == 1 && origin != []
+      number = origin[0]
+      origin.my_each do |item|
+        number = number.send(arg[0], item)
+      end
+      return number
+    end
+
+    if !block_given? && arg.length == 2 && origin != []
+      number = arg[0]
+      origin.my_each do |item|
+        number = number.send(arg[1], item)
+      end
+      return number
+    end
+
+    if block_given? && arg.length == 1 && origin != []
+      number = arg[0]
+      origin.my_each do |item|
+        number = yield(number, item)
+      end
+      return number
+    end
+
+    if block_given? && arg.length.zero? && origin != []
+      number = origin[0]
+      origin.my_each do |item|
+        number = yield(number, item)
+      end
     end
     number
   end
```

Comment:
```rb
2.6.5 :052 > array
 => [7, 2, 6, 1, 0, 7, 3, 7, 1, 4, 2, 4, 6, 5, 5, 8, 4, 0, 2, 7, 6, 2, 6, 2, 6, 1, 3, 7, 0, 4, 6, 8, 1, 5, 8, 8, 0, 7, 8, 3, 8, 8, 4, 8, 7, 2, 8, 3, 6, 3, 6, 1, 7, 1, 7, 5, 8, 7, 2, 6, 2, 6, 2, 6, 7, 8, 7, 8, 1, 5, 3, 5, 2, 1, 8, 6, 2, 7, 1, 8, 4, 6, 5, 3, 3, 5, 8, 7, 4, 0, 2, 3, 4, 2, 1, 0, 4, 6, 6, 4] 
2.6.5 :053 > array.inject
Traceback (most recent call last):
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `<main>'
        7: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/ruby_executable_hooks:24:in `eval'
        6: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `<main>'
        5: from /home/maelfosso/.rvm/gems/ruby-2.6.5/bin/irb:23:in `load'
        4: from /home/maelfosso/.rvm/gems/ruby-2.6.5/gems/irb-1.2.4/exe/irb:11:in `<top (required)>'
        3: from (irb):53
        2: from (irb):53:in `inject'
        1: from (irb):53:in `each'
LocalJumpError (no block given)
2.6.5 :054 > array.my_inject
 => nil 
2.6.5 :055 > 

```

Must raised a **LocalJumpError**

Reply:
```rb
2.6.5 :056 > operation = proc { |sum, n| sum + n }
2.6.5 :057 > array.inject(&operation)
 => 451 
2.6.5 :058 > array.my_inject(&operation)
 => 458 
2.6.5 :059 >
```

Reply:
```rb
2.6.5 :061 > array
 => [7, 2, 6, 1, 0, 7, 3, 7, 1, 4, 2, 4, 6, 5, 5, 8, 4, 0, 2, 7, 6, 2, 6, 2, 6, 1, 3, 7, 0, 4, 6, 8, 1, 5, 8, 8, 0, 7, 8, 3, 8, 8, 4, 8, 7, 2, 8, 3, 6, 3, 6, 1, 7, 1, 7, 5, 8, 7, 2, 6, 2, 6, 2, 6, 7, 8, 7, 8, 1, 5, 3, 5, 2, 1, 8, 6, 2, 7, 1, 8, 4, 6, 5, 3, 3, 5, 8, 7, 4, 0, 2, 3, 4, 2, 1, 0, 4, 6, 6, 4] 
2.6.5 :062 > array.inject(:+)
 => 451 
2.6.5 :063 > array.my_inject(:+)
 => 458 
2.6.5 :064 >
```

Reply:
```rb
2.6.5 :065 > range
 => 5..50 
2.6.5 :066 > range.my_inject(:*)
 => 6336269417023620425752626701263493509245341993533440000000000000 
2.6.5 :067 > range.inject(:*)
 => 1267253883404724085150525340252698701849068398706688000000000000 
2.6.5 :068 >
```

Reply:
We added all requested changes, except the LocalJumpError. Could you guys help us what that one and tell us how to rise the exception? Thank you.